### ISSUE TITLE
LambdaExpressions must not be compiled and executed

### ISSUE THREAD

**Issue by baseTwo** [FirelyTeam/firely-cql-sdk#617]
The output from the ExpressionBuilder returns a definition dictionary with LambdaExpressions. While it is possible to compile and invoke these dynamically, this must be disallowed. Critical fixes in the next step (expressions -> c#) is missed out on.

Only after the c# is compiled to assembly binaries, should the code be expected to get invoked.

**Comment by baseTwo** [comment]
Triaged against `develop` at `3810aa572`. **Done — and done structurally rather than by policy: dynamically compiling and invoking the compiler's output is no longer expressible.**

### The premise no longer holds

> "The output from the ExpressionBuilder returns a definition dictionary with LambdaExpressions."

Neither half of that is true any more.

**`ExpressionBuilder` is gone.** `grep -rn "class ExpressionBuilder" --include=*.cs Cql/` → 0 hits. It became `CodeBuilderContext` / `LibraryCodeBuilder` / `LibrarySetCodeBuilder`.

**The definition dictionary no longer holds `LambdaExpression`.** It holds `CqlDefinition` objects whose bodies are custom IR nodes — `Cql/Cql.Compiler/CqlDefinition.cs:135`:
```csharp
internal sealed class CqlDefinitionDictionary : DefinitionDictionary<CqlDefinition>;
```
and `Cql/Cql.Compiler/CqlDefinition.cs:28-39`:
```csharp
internal abstract class CqlLambdaDefinition(
    CodeLambda lambda,
    string name) : CqlDefinition(name)
{
    public CodeLambda Lambda { get; } = lambda;
```

`CodeLambda` (`Cql/Cql.Compiler/CodeModel/CodeLambda.cs:18-33`) is a plain IR node with `Parameters` and `Body` — it derives from `CodeExpression`, not from `System.Linq.Expressions.Expression`, and therefore **has no `.Compile()`**. It touches `System.Linq.Expressions` only to borrow `Expression.GetFuncType` for a type computation (`:23`). The emitter prints it as a C# local function.

The old delegate-lookup path is explicitly recorded as removed at `Cql/Cql.Compiler/CodeBuilderContext.Refs.cs:252-255`:
> "NOTE(phase4): the old builder built a `Func<>` delegate type here … purely to construct `FunctionCallExpression`, which needed it to reduce to a runtime `DefinitionDictionary<Delegate>` lookup. `CodeDefinitionCall` takes the return type directly and needs no delegate type."

### The only path to execution is now a compiled assembly

Exactly what the description asked for ("only after the c# is compiled to assembly binaries should the code be expected to get invoked"):

- `Cql/CodeGeneration.NET/AssemblyCompiler.cs:213-226` — Roslyn compiles the emitted C# and `Emit`s a real assembly.
- `Cql/Cql.Invocation/Toolkit/InvocationToolkit.cs:146` — those bytes are loaded into an `AssemblyLoadContext`; `LibraryInvoker`/`DefinitionInvoker` reflect over the loaded types (`Cql/Cql.Invocation/Toolkit/LibraryInvoker.cs:112`, `:129`).
- The generated assemblies cannot even reference expression trees: `Cql/CodeGeneration.NET/AssemblyCompiler.cs:28-29` deliberately omits the reference —
  > "`System.Linq.Expressions.dll` is intentionally omitted: generated C# no longer uses expression trees — the IR pipeline emits plain C# methods and delegates directly."

  Which means the prohibition is now enforced by the compilation itself, not by convention. That is a stronger outcome than "must be disallowed".

So the concern in the description — that "critical fixes in the next step (expressions → c#) are missed out on" by short-circuiting to a dynamically compiled lambda — is unreachable: there is no lambda to compile, and the IR → C# → assembly steps are the only route.

### One residual, not worth keeping this open for

`CqlContext.Definitions` is still declared as a `DefinitionDictionary<Delegate>` on the public runtime surface (`Cql/Cql.Runtime/Runtime/CqlContext.cs:51`, alias at `Cql/Cql.Runtime/GlobalUsings.cs:38`) — a leftover from the old interpreted path. Nothing in the SDK writes to it and nothing reads it; the real definition lookup is `LibraryInvoker.Definitions`, typed `IReadOnlyDictionary<DefinitionSignature, DefinitionInvoker>` (`Cql/Cql.Invocation/Toolkit/LibraryInvoker.cs:112`). Removing the dead member is a small public-API cleanup needing a `docs/releases/vnext/` fragment — worth its own ticket, but it is dead surface, not a live "compile and execute a LambdaExpression" path, so it does not keep this issue alive.

### If you grep for `.Compile()`, you will still find five hits — none of them is this issue

For the record, since this is the obvious objection: `Cql/Cql.Invocation/Toolkit/Internal/LibraryInvoker.5.0.cs:214`, `Cql/Cql.Runtime/Operators/CqlOperators.ClinicalOperators2.cs:106`, `Cql/Cql.Firely/CompiledPropertyAccessor.cs:65`, `Cql/Cql.Abstractions/Abstractions/Infrastructure/TypeExtensions.cs:155` and one test. Every one builds a delegate that calls an **already-compiled** method or a reflection/property accessor — `LibraryInvoker` does `Expression.Call(Expression.Constant(library), methodInfo, …)`, i.e. it invokes the loaded assembly. None of them compiles a CQL definition. That is precisely the order this issue asks for.

Closing as done.

Confidence: high — `ExpressionBuilder` has 0 matches, the definition dictionary is verifiably typed on `CqlDefinition`/`CodeLambda`, `CodeLambda` has no `Compile()`, and expression-tree support is deliberately excluded from the generated assemblies' references.
